radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/transport/handshake/AuthHandshaker.java

```diff
@@ -164,57 +164,64 @@ private AuthInitiateMessage createAuthInitiateMessage(ECPublicKey remotePubKey)
 
-		if (message.getNetworkId() != this.magic) {
-			return Pair.of(null, AuthHandshakeResult.error(
-				"Network ID mismatch",
-				Optional.of(NodeId.fromPublicKey(remotePubKey))
-			));
-		}
+			if (message.getNetworkId() != this.magic) {
+				return Pair.of(null, AuthHandshakeResult.error(
+					"Network ID mismatch",
+					Optional.of(NodeId.fromPublicKey(remotePubKey))
+				));
+			}
 
-		final var response = new AuthResponseMessage(
-			HashCode.fromBytes(ephemeralKey.getPublicKey().getBytes()),
-			HashCode.fromBytes(nonce)
-		);
-		final var encodedResponse = serialization.toDson(response, DsonOutput.Output.WIRE);
+			final var response = new AuthResponseMessage(
+				HashCode.fromBytes(ephemeralKey.getPublicKey().getBytes()),
+				HashCode.fromBytes(nonce)
+			);
+			final var encodedResponse = serialization.toDson(response, DsonOutput.Output.WIRE);
 
-		final var encryptedSize = encodedResponse.length + ECIESCoder.OVERHEAD_SIZE;
-		final var sizePrefix = ByteBuffer.allocate(2).putShort((short) encryptedSize).array();
-		final var encryptedResponsePayload =
-			ECIESCoder.encrypt(remotePubKey.getEcPoint(), encodedResponse, sizePrefix);
-		final var packet = new byte[sizePrefix.length + encryptedResponsePayload.length];
-		System.arraycopy(sizePrefix, 0, packet, 0, sizePrefix.length);
-		System.arraycopy(encryptedResponsePayload, 0, packet, sizePrefix.length, encryptedResponsePayload.length);
+			final var encryptedSize = encodedResponse.length + ECIESCoder.OVERHEAD_SIZE;
+			final var sizePrefix = ByteBuffer.allocate(2).putShort((short) encryptedSize).array();
+			final var encryptedResponsePayload =
+				ECIESCoder.encrypt(remotePubKey.getEcPoint(), encodedResponse, sizePrefix);
+			final var packet = new byte[sizePrefix.length + encryptedResponsePayload.length];
+			System.arraycopy(sizePrefix, 0, packet, 0, sizePrefix.length);
+			System.arraycopy(encryptedResponsePayload, 0, packet, sizePrefix.length, encryptedResponsePayload.length);
 
-		final var remoteEphemeralKey = extractEphemeralKey(
-			message.getSignature(),
-			message.getNonce(),
-			remotePubKey
-		);
+			final var remoteEphemeralKey = extractEphemeralKey(
+				message.getSignature(),
+				message.getNonce(),
+				remotePubKey
+			);
 
-		this.initiatePacketOpt = Optional.of(data);
-		this.responsePacketOpt = Optional.of(packet);
-		this.remotePubKeyOpt = Optional.of(remotePubKey);
+			this.initiatePacketOpt = Optional.of(data);
+			this.responsePacketOpt = Optional.of(packet);
+			this.remotePubKeyOpt = Optional.of(remotePubKey);
 
-		final var handshakeResult = finalizeHandshake(remoteEphemeralKey, message.getNonce());
-		return Pair.of(packet, handshakeResult);
+			final var handshakeResult = finalizeHandshake(remoteEphemeralKey, message.getNonce());
+			return Pair.of(packet, handshakeResult);
+		} catch (PublicKeyException | InvalidCipherTextException ex) {
+			return Pair.of(null, AuthHandshakeResult.error("Handshake decryption failed", Optional.empty()));
+		}
```

Comment:
It looks odd to return a `Result` while throwing `IOException`: callers need to consider both returns and exceptions.

A more elegant way of handling this would be to catch all `RuntimeException` and checked exceptions, and return an `Error`; otherwise, returns a `Success` result.

Reply:
yeah, I think it makes sense to catch IOException as well cause that's thrown by dson deserialization and indicates an invalid initiate msg. I wouldn't catch RuntimeExceptions though as they might be caused by anything really, not necessarily a faulty handshake message.
Alternatively we could model different errors with enums, but I think I'd rather just let RuntimeExceptions be handled by the upper layer.

Reply:
Sounds good!